app/vmalert/group.go

```diff
@@ -225,6 +236,22 @@ func (g *Group) start(ctx context.Context, querier datasource.Querier, nts []not
 	}
 }
 
+// updateWithTimeout tries to send update signal.
+// If timeout reached - it interrupts rules evaluation
+// and retries update attempt.
+func (g *Group) updateWithTimeout(ng *Group, timeout time.Duration) {
+	t := time.After(timeout)
+	select {
```

Comment:
add `defer t.Stop()` here in order tomis free up resources occupied by `t` before returning from this function

Reply:
That was intentional since I don't see how this could have noticeable impact. Even with frequent config reload it wont' exceed 1-2 calls per second.

Reply:
I see. But it is still better to properly clean up resources as soon as they aren't needed, since this code may be re-used later by the caller which could call it much more frequently, so the resource leak will become visible there.